master/buildbot/status/web/logs.py

```diff
@@ -117,41 +122,53 @@ def render_GET(self, req):
         else:
             req.setHeader("Cache-Control", "no-cache")
 
-        if not self.asText:
-            self.template = req.site.buildbot_service.templates.get_template("logs.html")
-            self.chunk_template = req.site.buildbot_service.templates.get_template("log_chunk.html")
-            builder = self.original.step.build.builder
-            build_id = self.original.step.build.number
-            url_dict = self.original.master.status.getURLForBuild(builder.getName(), build_id)
+        # If plaintext is requested just return the content of the logfile
+        if self.asText:
+            with_headers = "_with_headers" if self.withHeaders else ""
+            base_name = self.original.step.getName() + "_" + self.original.getName() + with_headers
+            base_name = base_name.replace(" ", "_") + ".txt"
```

Comment:
Watch out for forward slashes such as step 7 in http://katana.hq.unity3d.com/projects/Unity/builders/proj0-Build%20UserDocumentation%20-%20English%20%28w%2F%20PS3%29/builds/3066?unity_branch=5.4/staging

"Build UserDocumentation - English (w/ PS3)"